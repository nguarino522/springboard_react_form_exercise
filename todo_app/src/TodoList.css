.TodoList {
    color: white;
}

.TodoList-form-section {
    background-color: #2dd897c5;
    margin: auto;
    width: 40%;
    padding: 20px;
    border-radius: 15px;
}

.TodoList-task-section {
    background-color: #2dd897c5;
    margin-left: auto;
    margin-right: auto;
    margin-top: 20px;
    width: 40%;
    padding: 20px;
    border-radius: 15px;
    text-align: left;
}

ul {
    text-align: center;
    list-style-position: inside;
}