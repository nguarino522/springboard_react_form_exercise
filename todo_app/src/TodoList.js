import React, { useState } from 'react';
import './TodoList.css';
import Todo from './Todo';
import NewTodoForm from './NewTodoForm';

const TodoList = () => {
    const [todos, setTodos] = useState([]); 

    const createTodo = (newTodo) => {
        setTodos(todos => [...todos, newTodo]);
    }

    const removeTodo = (id) => {
        setTodos(todos => todos.filter(todo => todo.id !== id));
        console.log('fucking work')
    }

    return (
        <div className="TodoList">
            <div className="TodoList-form-section">
                <NewTodoForm createTodo={createTodo}/>
            </div>
            <div className="TodoList-task-section">
                <ul>
                    {todos.map( ({id, task}) => {
                        return <Todo key={id} id={id} task={task} removeTodo={removeTodo}/>
                    })}
                </ul>
            </div>
        </div>
    )
}

export default TodoList;
