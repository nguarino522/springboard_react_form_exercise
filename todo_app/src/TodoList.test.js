import React from "react";
import { render, fireEvent } from "@testing-library/react";
import TodoList from "./TodoList";

function addTodo(todoList, task = "write tests") {
    const taskInput = todoList.getByLabelText("Add a task todo:");
    fireEvent.change(taskInput, { target: { value: task } });
    const submitButton = todoList.getByText("Add Entry");
    fireEvent.click(submitButton);
}

it("renders without crashing", function () {
    render(<TodoList />);
});

it("matches snapshot", function () {
    const { asFragment } = render(<TodoList />);
    expect(asFragment()).toMatchSnapshot();
});

it("can add a todo", function () {
    const list = render(<TodoList />);
    addTodo(list);
    expect(list.getByLabelText("Add a task todo:")).toHaveValue("");
    expect(list.getByText("write tests")).toBeInTheDocument();
    expect(list.getByText("X")).toBeInTheDocument();
});

it("can delete a todo", function () {
    const list = render(<TodoList />);
    addTodo(list);
    expect(list.queryByText("write tests")).toBeInTheDocument();
    fireEvent.click(list.getByText("X"));
    expect(list.queryByText("write tests")).not.toBeInTheDocument();
});