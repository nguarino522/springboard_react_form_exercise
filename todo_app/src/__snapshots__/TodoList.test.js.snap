// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="TodoList"
  >
    <div
      class="TodoList-form-section"
    >
      <div
        class="NewTodoForm"
      >
        <form>
          <div
            class="NewTodoForm-section"
          >
            <label
              for="task"
            >
              Add a task todo: 
            </label>
            <input
              id="task"
              name="task"
              type="text"
              value=""
            />
          </div>
          <button>
            Add Entry
          </button>
        </form>
      </div>
    </div>
    <div
      class="TodoList-task-section"
    >
      <ul />
    </div>
  </div>
</DocumentFragment>
`;
